packages/@react-spectrum/s2/src/ActionButton.tsx

```diff
@@ -256,6 +259,7 @@ function ActionButton(props: ActionButtonProps, ref: FocusableRef<HTMLButtonElem
     size = props.size || 'M',
     isDisabled = props.isDisabled
   } = useSlottedContext(ActionButtonGroupContext) || {};
+  let ctx = !!useContext(ActionButtonGroupContext);
```

Comment:
```
let ctx = useSlottedContext...
let isInGroup = !!ctx
let {
  density...
} = ctx || {};
```
can shorten number of use context calls and prevent accidental usage in a slot situation